app/src/server/api/routers/payments.router.ts

```diff
@@ -0,0 +1,173 @@
+import { z } from "zod";
+import { createTRPCRouter, protectedProcedure } from "~/server/api/trpc";
+import { requireIsProjectAdmin, requireNothing } from "~/utils/accessControl";
+import { error, success } from "~/utils/errorHandling/standardResponses";
+import { TRPCError } from "@trpc/server";
+import {
+  createStripeCustomerAndConnectItToProject,
+  getDefaultPaymentMethodId,
+  getPaymentMethods,
+  getStripeCustomerId,
+  setDefaultPaymentMethod,
+  createSetupIntent,
+  getStripeCustomer,
+  deletePaymentMethod,
+} from "~/server/utils/stripe";
+import { chargeInvoice } from "~/server/tasks/chargeInvoices.task";
+
+export const paymentsRouter = createTRPCRouter({
+  createStripeCustomer: protectedProcedure
+    .input(
+      z.object({
+        projectId: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      await requireIsProjectAdmin(input.projectId, ctx);
+
+      const stripeCustomerId = await getStripeCustomerId({
+        projectId: input.projectId,
+        required: false,
+      });
+
+      if (stripeCustomerId) {
+        return success("Stripe customer already created.");
+      }
+
+      try {
+        await createStripeCustomerAndConnectItToProject(input.projectId);
+        return success("Stripe customer created successfully!");
+      } catch {
+        return error("Failed to create a stripe customer.");
+      }
+    }),
+  createStripeIntent: protectedProcedure
+    .input(
+      z.object({
+        projectId: z.string(),
+      }),
+    )
+    .mutation(async ({ input, ctx }) => {
+      await requireIsProjectAdmin(input.projectId, ctx);
+
+      let stripeCustomerId = await getStripeCustomerId({
+        projectId: input.projectId,
+        required: true,
+      });
+
+      // Check if stripe user was deleted. If yes - create a new one
```

Comment:
Under what circumstances will the stripe user be deleted?

Reply:
A user can be deleted through the stripe dashboard. In this case, it would throw an error.
A stripe user has a type of `Stripe.Response<Stripe.Customer | Stripe.DeletedCustomer>`.